bpf/tests/wireguard_helpers_tests.c

```diff
@@ -93,3 +97,41 @@ int check2(struct __ctx_buff *ctx)
 
 	test_finish();
 }
+
+PKTGEN("tc", "wireguard_icmpv6_na_skip")
+int wireguard_icmpv6_na_skip_pktgen(struct __ctx_buff *ctx)
+{
+	struct pktgen builder;
+	struct icmp6hdr *icmp6;
+
+	pktgen__init(&builder, ctx);
+
+	icmp6 = pktgen__push_ipv6_icmp6_packet(&builder,
+					       (__u8 *)mac_one,
+					       (__u8 *)mac_two,
+					       (__u8 *)v6_pod_one,
+					       (__u8 *)v6_pod_two,
+					       ICMPV6_NA_MSG);
+	if (!icmp6)
+		return TEST_ERROR;
+
+	pktgen__finish(&builder);
+
+	return 0;
+}
+
+CHECK("tc", "wireguard_icmpv6_na_skip")
+int wireguard_icmpv6_na_skip_check(struct __ctx_buff *ctx)
+{
+	int ret_val;
+
+	ipcache_v6_add_entry((union v6addr *)v6_pod_one, 0, 123, 0, 1);
+	ipcache_v6_add_entry((union v6addr *)v6_pod_two, 0, 456, 0, 1);
+
+	test_init();
+
+	ret_val = wg_maybe_redirect_to_encrypt(ctx, bpf_htons(ETH_P_IPV6), 123);
+	assert(ret_val == CTX_ACT_OK);
+
+	test_finish();
+}
```

Comment:
I fairly randomly placed the test here, I bet we can find a better place for it in the future :slightly_smiling_face:

Reply:
I'd like to soon phase out the manual packet parsing logic in `ctx_is_wireguard` (tested here) in favor of the new classifiers I'm adding. We could rename this file to `wireguard_redirect_tests.c` and add more bpf unit testing as we're doing for the `ipsec_redirect_{native,tunnel}`.

That can for sure happen in a future PR, just want to make sure we all agree 😃